tools/ci-ts/test-helpers/common.ts

```diff
@@ -162,3 +200,61 @@ export async function fundAddress(to: string, ether = 1000) {
   })
   await tx.wait()
 }
+
+/**
+ * helper function to more seamlessly wait for transactions to confirm
+ * before continuing with test execution
+ * @param tx transaction to wait for
+ */
+export async function txWait(
+  tx: ethers.ContractTransaction,
+): Promise<ContractReceipt> {
+  return await tx.wait()
+}
+
+/**
+ * adds a listener to the provided contract to watch for the provided
+ * events; automatically parses the log data, logs the event emissions,
+ * and logs the values of the event data
+ * @param contract contract to watch for events on
+ * @param _listenTo event or list of events to listen to
+ */
+export async function logEvents(
+  contract: ethers.Contract,
+  _listenTo: string | string[] = [],
+) {
+  const listenTo = Array.isArray(_listenTo) ? _listenTo : [_listenTo]
+  assert.containsAllKeys(
+    contract.interface.events,
+    listenTo,
+    'contract does not have requested event type',
+  )
+
+  // holds the contract's events, keys are the sha256 of the event signature
+  const eventsByTopic: {
+    [key: string]: EventDescription
+  } = Object.entries(contract.interface.events).reduce((prev, [_, event]) => {
+    return { [event.topic]: event, ...prev }
+  }, {})
+
+  contract.on('*', (...args) => {
```

Comment:
This has a ton of logic contained within it, can we use some unit tests for it?

Reply:
It feels kinds weird to write tests for test-helper functions. Where should the tests live? When do they get run?

Reply:
You could put them right beside the file, `common.test.ts`. They would be run as a separate part of CI, since they are unit tests. At least from personal experience, it will save a lot of debugging pain if complex test helpers get their own tests, so you can have confidence that the actual systems under test are failing, rather than your test helpers.

It's a lot faster to fix unit tests than integration tests.